plugins/azure-devops-common/src/types.ts

```diff
@@ -126,3 +126,128 @@ export type PullRequestOptions = {
   top: number;
   status: PullRequestStatus;
 };
+
+export interface DashboardPullRequest {
+  pullRequestId?: number;
+  title?: string;
+  description?: string;
+  repository?: Repository;
+  createdBy?: CreatedBy;
+  hasAutoComplete: boolean;
+  policies?: Policy[];
+  reviewers?: Reviewer[];
+  creationDate?: string;
+  status?: PullRequestStatus;
+  isDraft?: boolean;
+  link?: string;
+}
+
+export interface Reviewer {
+  id?: string;
+  displayName?: string;
+  imageUrl?: string;
+  isRequired?: boolean;
+  isContainer?: boolean;
+  voteStatus: PullRequestVoteStatus;
+}
+
+export interface Policy {
+  id?: number;
+  type: PolicyType;
+  status?: PolicyEvaluationStatus;
+  text?: string;
+  link?: string;
+}
+
+export interface CreatedBy {
+  id?: string;
+  displayName?: string;
+  uniqueName?: string;
+  imageUrl?: string;
+}
+
+export interface Repository {
+  id?: string;
+  name?: string;
+  url?: string;
+}
+
+export interface Team {
+  id?: string;
+  name?: string;
+  memberIds?: string[];
+}
+
+/**
+ * Status of a policy which is running against a specific pull request.
+ */
+export enum PolicyEvaluationStatus {
+  /**
+   * The policy is either queued to run, or is waiting for some event before progressing.
+   */
+  Queued = 0,
+  /**
+   * The policy is currently running.
+   */
+  Running = 1,
+  /**
+   * The policy has been fulfilled for this pull request.
+   */
+  Approved = 2,
+  /**
+   * The policy has rejected this pull request.
+   */
+  Rejected = 3,
+  /**
+   * The policy does not apply to this pull request.
+   */
+  NotApplicable = 4,
+  /**
+   * The policy has encountered an unexpected error.
+   */
+  Broken = 5,
+}
+
+export enum PolicyType {
+  Build = 'Build',
+  Status = 'Status',
+  MinimumReviewers = 'MinimumReviewers',
+  Comments = 'Comments',
+  RequiredReviewers = 'RequiredReviewers',
+  MergeStrategy = 'MergeStrategy',
+}
+
+export const PolicyTypeId = {
```

Comment:
Just because I'm curious, how come this one doesn't get to be an enum but the other ones do? xP

Reply:
I didn't know if it would map correctly in this:

```ts
const policyType: PolicyType | undefined = (
  {
    [PolicyTypeId.Build]: PolicyType.Build,
    [PolicyTypeId.Status]: PolicyType.Status,
    [PolicyTypeId.MinimumReviewers]: PolicyType.MinimumReviewers,
    [PolicyTypeId.Comments]: PolicyType.Comments,
    [PolicyTypeId.RequiredReviewers]: PolicyType.RequiredReviewers,
    [PolicyTypeId.MergeStrategy]: PolicyType.MergeStrategy,
  } as Record<string, PolicyType | undefined>
)[policyTypeId];
```

Turns out it's fine so I will convert to enum.